### PR TITLE
Exception as event

### LINKED ISSUE #152: RecordException doesn't capture Exception as Event
**Describe the bug**
`.recordException` does not send exception as Event

**Steps to reproduce**
Throw an Error and try to recordExpection on it

**What did you expect to see?**
It to create this as an event

**What did you see instead?**
No Event

**What version and what artifacts are you using?**
Artifacts: https://pub.dev/packages/opentelemetry
Version: 0.18.0
How did you reference these artifacts? (excerpt from your `pubspec.lock`, `pubspec.yaml`, etc)

**Environment**
Dart Version:  3.2.3
OS: Windows 11

**Additional context**
There is still a TODO on this function, that should be implemented. According to the Opentelemetry standard these need to be an event with the name: "exception" https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/
https://github.com/Workiva/opentelemetry-dart/blob/fe771987782fb7aa26f18980e833075f30b26a00/lib/src/sdk/trace/span.dart#L140-L152

### PR DESCRIPTION
**Pull request by mikebosland**
## Which problem is this PR solving?

Fix a TODO to correct `recordException` as an event instead of attributes directly on the span. See the spec: https://opentelemetry.io/docs/specs/otel/trace/exceptions/

Fixes #152 (issue)

## Short description of the change

Implement as an event instead of span attributes.

## How Has This Been Tested?

Adjusted existing integration test.

## Checklist:

- [X] Unit tests have been added
- [X] Documentation has been updated

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/unit/mocks.mocks.dart [hunk 1]
```diff
@@ -1,6 +1,3 @@
-// Copyright 2021-2022 Workiva.
-// Licensed under the Apache License, Version 2.0. Please see https://github.com/Workiva/opentelemetry-dart/blob/master/LICENSE for more information
```
Comment (blakeroberts-wk) [inline test/unit/mocks.mocks.dart:2]: could you run `make format` to reattach this comment?

Comment (mikebosland) [reply test/unit/mocks.mocks.dart]: I tried `make format`. It added the copyright header to 116 files. Apparently the logic to detect pre-existing copyright headers isn't working...  I've restored the header for this file directly.

**Comment by blakeroberts-wk** [comment]
@mikebosland could you regenerate the mocks?

**Comment by mikebosland** [comment]
> @mikebosland could you regenerate the mocks?

Hi, I just went to regenerate the mocks. The only diff was to remove the copyright header, so I suppose they are to date, unless I'm missing something...

**Comment by blakeroberts-wk** [comment]
@mikebosland I think you will need to merge master and regenerate the mocks to resolve the merge conflicts?

**Comment by mikebosland** [comment]
Okay, I think I've merged the master. Running the command to regenerate the mocks just removes the file, but all tests continue to pass.

**Review by blakeroberts-wk — APPROVED** [review]

**Comment by blakeroberts-wk** [comment]
QA +1

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**